assets/src/components/admin-notice/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { VisuallyHidden } from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+export const ADMIN_NOTICE_TYPE_INFO = 'info';
+export const ADMIN_NOTICE_TYPE_SUCCESS = 'success';
+export const ADMIN_NOTICE_TYPE_WARNING = 'warning';
+export const ADMIN_NOTICE_TYPE_ERROR = 'error';
+
+/**
+ * WordPress admin notice.
+ *
+ * @param {Object}   props               Component props.
+ * @param {Element}  props.children      Component children.
+ * @param {string}   props.className     Additional class names.
+ * @param {boolean}  props.isDismissible Indicates whether the notice should be dismissible.
+ * @param {Function} props.onDismiss     Function to be called whenever the notice gets dismissed.
+ * @param {string}   props.type          Specifies type of the notice.
+ */
+export function AdminNotice( {
+	children,
+	className,
+	isDismissible = false,
+	onDismiss,
+	type = ADMIN_NOTICE_TYPE_INFO,
+} ) {
+	const [ dismissed, setDismissed ] = useState( false );
+
+	const dismissHandler = useCallback( () => {
+		setDismissed( true );
+
+		if ( typeof onDismiss === 'function' ) {
+			onDismiss();
+		}
+	}, [ onDismiss ] );
+
+	if ( isDismissible && dismissed ) {
+		return null;
+	}
+
+	return (
+		<div
+			className={ classnames( 'admin-notice', className, {
+				'admin-notice--dismissible': isDismissible,
+				'admin-notice--info': type === ADMIN_NOTICE_TYPE_INFO,
+				'admin-notice--success': type === ADMIN_NOTICE_TYPE_SUCCESS,
+				'admin-notice--warning': type === ADMIN_NOTICE_TYPE_WARNING,
+				'admin-notice--error': type === ADMIN_NOTICE_TYPE_ERROR,
```

Comment:
As noted in the stylesheet, these class names should all have an `amp-` prefix, right?

```suggestion
			className={ classnames( 'amp-admin-notice', className, {
				'amp-admin-notice--dismissible': isDismissible,
				'amp-admin-notice--info': type === ADMIN_NOTICE_TYPE_INFO,
				'amp-admin-notice--success': type === ADMIN_NOTICE_TYPE_SUCCESS,
				'amp-admin-notice--warning': type === ADMIN_NOTICE_TYPE_WARNING,
				'amp-admin-notice--error': type === ADMIN_NOTICE_TYPE_ERROR,
```

Reply:
Question: Any reason why core's class names couldn't be re-used? The core-provided activation notice is:

```html
<div id="message" class="updated notice is-dismissible">
    <p>Plugin activated.</p>
    <button type="button" class="notice-dismiss">
    <span class="screen-reader-text">Dismiss this notice.</span></button>
</div>
```

It seems like notice could use the same class names as `notice` and either `notice-info`, `notice-warning`, or `notice-error`?

Reply:
The core notice component was my starting point. It turned out, however, that marrying the [existing jQuery-based logic](https://github.com/WordPress/WordPress/blob/master/wp-admin/js/common.js#L1085-L1116) with the React is quite cumbersome. Since the notice is pretty simple, I decided to recreate it as a React component.

Reply:
The component is now renamed to `AmpAdminNotice` (0f2c38a).